main.go

```diff
@@ -24,15 +25,25 @@ func main() {
 		os.Exit(1)
 	}
 
-	// input send text
+	//Form JSON payload to send to Slack
+	var json string
 	args := strings.Join(os.Args[1:], " ")
 	if args == "" {
-		fmt.Fprintln(os.Stdout, "usage: slack-notifier <TEXT>")
-		os.Exit(0)
+		// use stdin as post data
+		if buf, err := ioutil.ReadAll(os.Stdin); err != nil {
+			fmt.Fprintln(os.Stderr, err)
+			os.Exit(1)
+		} else if len(buf) > 0 {
+			json = string(buf)
+		} else {
+			fmt.Fprintln(os.Stderr, "usage: slack-notifier <TEXT>")
+			os.Exit(1)
+		}
```

Comment:
This can be written more simply:

```go
		if buf, err := ioutil.ReadAll(os.Stdin); err != nil {
			fmt.Fprintln(os.Stderr, err)
			os.Exit(1)
		}

		if len(buf) == 0 {
			fmt.Fprintln(os.Stderr, "usage: slack-notifier <TEXT>")
			os.Exit(1)
		}

		json = string(buf)
```